### PR TITLE
tiles: fix off by one in dcache bounds check

### PR DESCRIPTION
**Pull request by mmcgee-jump**

### CODE REVIEW

File: src/disco/dedup/fd_dedup.c [hunk 1]
```diff
@@ -51,7 +51,7 @@ during_frag( void * _ctx,
 
   fd_dedup_ctx_t * ctx = (fd_dedup_ctx_t *)_ctx;
 
-  if( FD_UNLIKELY( chunk<ctx->in[ in_idx ].chunk0 || chunk>=ctx->in[ in_idx ].wmark || sz > FD_TPU_DCACHE_MTU ) )
+  if( FD_UNLIKELY( chunk<ctx->in[ in_idx ].chunk0 || chunk>ctx->in[ in_idx ].wmark || sz > FD_TPU_DCACHE_MTU ) )
     FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu)", chunk, sz, ctx->in[ in_idx ].chunk0, ctx->in[ in_idx ].wmark ));
```
Comment (ptaffet-jump) [inline src/disco/dedup/fd_dedup.c:55]:
```suggestion
    FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->in[ in_idx ].chunk0, ctx->in[ in_idx ].wmark ));
```

File: src/disco/net/fd_net.c [hunk 2]
```diff
@@ -137,7 +137,7 @@ during_frag( void * _ctx,
 
   fd_net_ctx_t * ctx = (fd_net_ctx_t *)_ctx;
 
-  if( FD_UNLIKELY( chunk<ctx->in_chunk0 || chunk>=ctx->in_wmark || sz > FD_NET_MTU ) )
+  if( FD_UNLIKELY( chunk<ctx->in_chunk0 || chunk>ctx->in_wmark || sz > FD_NET_MTU ) )
     FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu)", chunk, sz, ctx->in_chunk0, ctx->in_wmark ));
```
Comment (ptaffet-jump) [inline src/disco/net/fd_net.c:141]:
```suggestion
    FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->in_chunk0, ctx->in_wmark ));
```

File: src/disco/pack/fd_pack.c [hunk 3]
```diff
@@ -93,7 +93,7 @@ during_frag( void * _ctx,
              int *  opt_filter ) {
   fd_pack_ctx_t * ctx = (fd_pack_ctx_t *)_ctx;
 
-  if( FD_UNLIKELY( chunk<ctx->in[ in_idx ].chunk0 || chunk>=ctx->in[ in_idx ].wmark || sz > FD_TPU_DCACHE_MTU ) )
+  if( FD_UNLIKELY( chunk<ctx->in[ in_idx ].chunk0 || chunk>ctx->in[ in_idx ].wmark || sz > FD_TPU_DCACHE_MTU ) )
     FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu)", chunk, sz, ctx->in[ in_idx ].chunk0, ctx->in[ in_idx ].wmark ));
```
Comment (ptaffet-jump) [inline src/disco/pack/fd_pack.c:97]:
```suggestion
    FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->in[ in_idx ].chunk0, ctx->in[ in_idx ].wmark ));
```

File: src/disco/quic/fd_quic.c [hunk 4]
```diff
@@ -333,7 +333,7 @@ during_frag( void * _ctx,
 
   fd_quic_ctx_t * ctx = (fd_quic_ctx_t *)_ctx;
 
-  if( FD_UNLIKELY( chunk<ctx->in_chunk0 || chunk>=ctx->in_wmark || sz > FD_NET_MTU ) )
+  if( FD_UNLIKELY( chunk<ctx->in_chunk0 || chunk>ctx->in_wmark || sz > FD_NET_MTU ) )
     FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu)", chunk, sz, ctx->in_chunk0, ctx->in_wmark ));
```
Comment (ptaffet-jump) [inline src/disco/quic/fd_quic.c:337]:
```suggestion
    FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->in_chunk0, ctx->in_wmark ));
```

File: src/disco/verify/fd_verify.c [hunk 5]
```diff
@@ -39,7 +39,7 @@ during_frag( void * _ctx,
     return;
   }
 
-  if( FD_UNLIKELY( chunk<ctx->in[in_idx].chunk0 || chunk>=ctx->in[in_idx].wmark || sz > FD_TPU_DCACHE_MTU ) )
+  if( FD_UNLIKELY( chunk<ctx->in[in_idx].chunk0 || chunk>ctx->in[in_idx].wmark || sz > FD_TPU_DCACHE_MTU ) )
     FD_LOG_WARNING(( "chunk %lu %lu corrupt, not in range [%lu,%lu)", chunk, sz, ctx->in[in_idx].chunk0, ctx->in[in_idx].wmark ));
```
Comment (ptaffet-jump) [inline src/disco/verify/fd_verify.c:43]:
```suggestion
    FD_LOG_WARNING(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz, ctx->in[in_idx].chunk0, ctx->in[in_idx].wmark ));
```

Comment (ptaffet-jump) [reply src/disco/verify/fd_verify.c:43]: Why is this one just a warning?

Comment (mmcgee-jump) [reply src/disco/verify/fd_verify.c]: QUIC can overrun frags, which could cause a corrupt chunk if we are unlucky

**Review by ptaffet-jump — APPROVED** [review]

**Pull request merged**